tests/unittests/std_interoperability.cpp

```diff
@@ -403,33 +411,205 @@ TEST_CASE("Collection AllocatorAwareContainer types", "[collection][container][t
+    // std::decay_t<const_iterator::value_type::mutable_type>>);
+    // {
+    //   auto coll = CollectionType{};
+    //   coll.create().cellID(42);
+    //   auto e = iterator::value_type::mutable_type{13, 0, 0, 0, 0};
+    //   auto e_copy = e;
+    //   auto o = coll.cbegin();
+    //   *o = e;
+    //   REQUIRE(*o == e_copy);
+    // }
+
+    // iterator
+    DOCUMENTED_STATIC_FAILURE(std::output_iterator<iterator, CollectionType::value_type>);
+    DOCUMENTED_STATIC_FAILURE(std::output_iterator<iterator, CollectionType::value_type::mutable_type>);
+    // const_iterator
     DOCUMENTED_STATIC_FAILURE(std::output_iterator<const_iterator, CollectionType::value_type>);
+    DOCUMENTED_STATIC_FAILURE(std::output_iterator<const_iterator, CollectionType::value_type::mutable_type>);
+  }
+
+  SECTION("forward_iterator") {
+    // iterator
+    DOCUMENTED_STATIC_FAILURE(std::forward_iterator<iterator>);
+    // {
+    //   REQUIRE(iterator{} == iterator{});
+    //   auto coll = CollectionType();
+    //   coll.create();
+    //   auto i = coll.begin();
+    //   auto j = coll.begin();
+    //   REQUIRE(i == j);
+    //   REQUIRE(++i == ++j);
+    //   i = coll.begin();
+    //   REQUIRE(((void)[](auto x) { ++x; }(i), *i) == *i);
+    //   Pointers and references obtained from a forward iterator into a range remain valid while the range exists.
+    //   Is this even unit-testable?
+    // }
+    // const_iterator
     DOCUMENTED_STATIC_FAILURE(std::forward_iterator<const_iterator>);
+    // {
+    //   REQUIRE(const_iterator{} == const_iterator{});
+    //   auto coll = CollectionType();
+    //   coll.create();
+    //   auto i = coll.cbegin();
+    //   auto j = coll.cbegin();
+    //   REQUIRE(i == j);
+    //   REQUIRE(++i == ++j);
+    //   i = coll.cbegin();
+    //   REQUIRE(((void)[](auto x) { ++x; }(i), *i) == *i);
+    //   Pointers and references obtained from a forward iterator into a range remain valid while the range exists.
+    //   Is this even unit-testable?
+    // }
+  }
+
+  SECTION("bidirectional_iterator") {
+    // iterator
+    DOCUMENTED_STATIC_FAILURE(std::bidirectional_iterator<iterator>);
+    // TODO check semantic requirements
+    // const_iterator
     DOCUMENTED_STATIC_FAILURE(std::bidirectional_iterator<const_iterator>);
+    // TODO check semantic requirements
+  }
+
+  SECTION("random_access_iterator") {
+    // iterator
+    DOCUMENTED_STATIC_FAILURE(std::random_access_iterator<iterator>);
+    // TODO check semantic requirements
+    // const_iterator
     DOCUMENTED_STATIC_FAILURE(std::random_access_iterator<const_iterator>);
+    // TODO check semantic requirements
+  }
+
+  SECTION("contiguous_iterator_iterator") {
+    // iterator
+    DOCUMENTED_STATIC_FAILURE(std::contiguous_iterator<iterator>);
+    // TODO check semantic requirements
+    // const_iterator
     DOCUMENTED_STATIC_FAILURE(std::contiguous_iterator<const_iterator>);
+    // TODO check semantic requirements
```

Comment:
Can we group the ones where we already know that we have no chance of getting them with the current design into one section and label that accordingly?